all-actors/src/main/java/org/sunbird/serviceimpl/CertsServiceImpl.java

```diff
@@ -174,4 +175,49 @@ public Response generate(Request request) throws BaseException {
         }
         return response;
     }
+
+    @Override
+    public Response verify(Request request) throws BaseException {
+        Response response = new Response();
+        try {
+            Map<String, Object> certVerifyReqMap = new HashMap<>();
+            certVerifyReqMap.put(JsonKeys.REQUEST,composeCertVerifyRequest(request));
+            String requestBody = requestMapper.writeValueAsString(certVerifyReqMap);
+            logger.info("CertsServiceImpl:verify:request body prepared:" + requestBody);
+            String apiToCall = CertVars.getSERVICE_BASE_URL().concat(CertVars.getVerifyUri());
+            logger.info("CertsServiceImpl:verify:complete url prepared:" + apiToCall);
+            Future<HttpResponse<JsonNode>>responseFuture=CertificateUtil.makeAsyncPostCall(apiToCall,requestBody,headerMap);
+            HttpResponse<JsonNode> jsonResponse = responseFuture.get();
+            if (jsonResponse != null && jsonResponse.getStatus() == HttpStatus.SC_OK) {
+                String stringifyResponse=jsonResponse.getBody().getObject().getJSONObject(JsonKeys.RESULT).get(JsonKeys.RESPONSE).toString();
+                Map<String,Object> apiResp=requestMapper.readValue(stringifyResponse,Map.class);
+                response.put(JsonKeys.RESPONSE,apiResp);
+            }
+            else if(jsonResponse!=null && jsonResponse.getStatus() == HttpStatus.SC_BAD_REQUEST){
+                String stringifyResponse=jsonResponse.getBody().getObject().getJSONObject(JsonKeys.RESULT).get(JsonKeys.MESSAGE).toString();
+                throw new BaseException(IResponseMessage.INVALID_REQUESTED_DATA,stringifyResponse, ResponseCode.CLIENT_ERROR.getCode());
+            }
+            else {
+                throw new BaseException(IResponseMessage.INVALID_REQUESTED_DATA,IResponseMessage.INVALID_REQUESTED_DATA, ResponseCode.CLIENT_ERROR.getCode());
+            }
+
+        } catch (Exception e) {
+            logger.error("CertsServiceImpl:verify:exception occurred:" + e);
+            throw new BaseException(IResponseMessage.INTERNAL_ERROR, e.getMessage(), ResponseCode.SERVER_ERROR.getCode());
+        }
+        return response;
+    }
+
+    private Map<String,Object> composeCertVerifyRequest(Request request){
+        Map<String,Object>certificate=new HashMap<>();
+        Map<String,Object>certVerifyMap=new HashMap<>();
+        if(MapUtils.isNotEmpty((Map)request.getRequest().get(JsonKeys.DATA))) {
+            certificate.put(JsonKeys.DATA, request.getRequest().get(JsonKeys.DATA));
+        }
+        if(StringUtils.isNotBlank((String)request.getRequest().get(JsonKeys.ID))) {
```

Comment:
mark this as else-if. What happens if some body sends you both?

Reply:
@indrajra if someone sends me both this will work